mapApp/static/mapApp/js/index.js

```diff
@@ -1,105 +1,167 @@
-// Create data feature groups
-var collisions, nearmisses, hazards, thefts, newInfrastructures;
-
-//'159.203.2.12' for dev
-var srv = window.location.hostname;
+/** Host location, port to construct API requests */
+var hostname = window.location.hostname;
 
 // If running locally, include port in API requests
 if (window.location.port) {
-  srv = srv + ':' + window.location.port;
+  hostname = hostname + ':' + window.location.port;
 }
 
-loadIncidenLayerXHR("/nearmisses_tiny?format=json", "nearmiss", nearmisses);
-loadIncidenLayerXHR("/hazards_tiny?format=json", "hazard", hazards);
-loadIncidenLayerXHR("/thefts_tiny?format=json", "theft", thefts);
-loadIncidenLayerXHR("/collisions_tiny?format=json", "collision", collisions);
-loadIncidenLayerXHR("/newInfrastructures_tiny?format=json", "newInfrastructure", newInfrastructures);
+/**
+* Map set up
+*/
 
-var xhrLayersLoaded = 0;
-var refLayers = [];
-
-var incidentData = new L.MarkerClusterGroup({
-    maxClusterRadius: 70,
-    polygonOptions: {
-        color: '#2c3e50',
-        weight: 3
-    },
-    animateAddingMarkers: true,
-    iconCreateFunction: pieChart
-}),
-    alertAreas = L.featureGroup();
-
-// Define popup getter function
-incidentData.on('click', function (e) {
-    var layer = e.layer;
-    //layer.bindPopup(getPopup(layer)).openPopup();
-    getXHRPopup(layer);
-});
+/** Initialize the map */
 
+// Check if map center and zoom are provided in url, otherwise default to view of north america
```

Comment:
It looks like anyone visiting https://bikemaps.org will start at the N, America extent. Since the bulk of our data is from N. America, I think the majority of the data will still be downloaded when the map first loads. I wonder if it would be possible to delay loading any data until:
a. The user declines geolocation
b. The user accepts (or has previously accepted geolocation) and the geolocation and setting of the map view is complete.
Thoughts?